pycaret/internal/ensemble.py

```diff
@@ -1,140 +1,129 @@
+        )
+
+    def _check_method(self):
+        if self.method == "voting" and self.weights is None:
+            warnings.warn(
+                "Missing 'weights' argument, setting uniform weights."
+            )
+            self._set_weights_uniform
+        elif self.method in self._not_required_weights and self.weights:
+            warnings.warn(
+                "Unused 'weights' argument. When method='mean' or method='median', 'weights' argument is not provided. Setting weights to `None`"
+            )
+            self._set_weights_none
+        elif self.method not in self._available_methods:
+            raise ValueError(
+                f"Method {self.method} is not supported. Avaible methods are {', '.join(self._available_methods)}"
+            )
+
+    def _check_weights(self):
+        if (self.weights is not None and len(self.weights) != len(self.forecasters)):
+            raise ValueError(
+                f"Number of forecasters and weights must be equal, got {len(self.weights)} weights and {len(self.estimators)} estimators"
+            )
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        names, forecasters = self._check_forecasters()
+        self._fit_forecasters(forecasters, y, X, fh)
+        return self
+
+    def update(self, y, X=None, update_params=True):
+        """Update fitted parameters
+        Parameters
+        ----------
+        y : pd.Series
+        X : pd.DataFrame
+        update_params : bool, optional (default=True)
+        Returns
+        -------
+        self : an instance of self
+        """
+        for forecaster in self.forecasters_:
+            forecaster.update(y, X, update_params=update_params)
+        return self
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        if return_pred_int:
+            raise NotImplementedError()
+
+        self._check_method()
+
+        pred_forecasters = pd.concat(self._predict_forecasters(fh, X), axis=1)
+
+        if self.method == "median":
+            return pred_forecasters.median(axis=1)
+        elif self.method in self._required_weights:
+            self._check_weights()
+            pred_w = np.average(pred_forecasters, axis=1, weights=self.weights)
+            return pd.DataFrame(pred_w, index=pred_forecasters.index)
+
+    @property
+    def _set_weights_none(self):
+        self.weights = None
+
+    @property
+    def _set_weights_uniform(self):
+        self.weights = np.ones(len(self.forecasters))
```

Comment:
Shouldn't those be methods? Is this how sktime does it?

Reply:
Sktime original implementation doesn't handle `voting` method, here I should be changing these to appropiate setter for `@property`

```python
@property
 def weights(self):
      return self.weights

@weights.setter
def weights(self, value):
    self.weights = value
```

and then call

```python
elif self.method in self._not_required_weights and self.weights:
        warnings.warn(
            "Unused 'weights' argument. When method='mean' or method='median', 'weights' argument is not provided. Setting weights to `None`"
        )
        self.weights = None
```

Reply:
@Yard1 I think we should open a PR to include `voting` in sktime original implementation, what are your thoughts on this ?

Reply:
That's a great idea @TremaMiguel